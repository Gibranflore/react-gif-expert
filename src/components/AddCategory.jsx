import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = ( event ) => { //Todo esto es la logica del formulario captura lo que escribimos
        event.preventDefault();
        if (inputValue.trim().length <=1) return; //El if llama al usatate.trim y que tenga almentos mas de un caracter

        //setCategories(categories=> [ inputValue, ...categories ]);
        setInputValue(''); //esto ayuda una vez que demos enter borre el formulario
        onNewCategory(inputValue.trim())//El trim ayuda a verificar lose espacios adelante y atras
    }

  return (
    <form onSubmit={(event)=> onSubmit(event)} aria-label='form'>
        <input
            type="text"
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={onInputChange}
        />
    </form>
    )

}
/* El return tien un formulario form que guarda un onSubmit
"on" se refirere que esta llamdo algo, en este caso un event "event"
es un funcion que esta llamndo en este caso el onSubmit */

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}