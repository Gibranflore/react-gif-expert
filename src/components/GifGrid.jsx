
import { GifItem } from "./GifItem";
import { useFetchGifs } from "./hooks/useFetchGifs";


export const GifGrid = ({category}) => {
    
    //react vuelve a disparar 2 veceses este componente

    const { images, isLoading } = useFetchGifs ( category );

    //getGifs( category ); //no volver a disparar la peticion nuevamente

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && ( <h2>Cargando...</h2>) //aparecer cargando cuando buscas
            }


            <div className="card-grid">
                {
                    images.map( ( image )=> (
                       <GifItem 
                            key={image.id}
                            {...image}
                       />

                    ))
                }
            
            </div>
        </>
    )
}
