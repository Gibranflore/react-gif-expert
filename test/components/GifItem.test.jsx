

import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Prueblas en GifItem', () => { 

    const title = 'Earth';
    const url = 'https://earth.com/live.jpg'


    test('Debe hacer match con el snapshot', () => { 

        const { container } = render ( <GifItem title={title} url={ url } /> )
        expect(container).toMatchSnapshot();

    });

    test('Debe mostrar el url y el ALT indicado', () => {

        //expect( screen.getByRole('img').alt ).toBe( title + "!" );
        //expect( screen.getByRole('img').rsc ).toBe( url );
        render ( <GifItem title={title} url={ url } /> )

        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('Debe de mostrar el titulo en el componente', () => { 
        render ( <GifItem title={title} url={ url } /> )
        expect( screen.getByText(title) ).toBeTruthy(); //En el screen ver si existe
    })
}); 