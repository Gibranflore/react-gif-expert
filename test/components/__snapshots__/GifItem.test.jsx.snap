// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueblas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    GifItem
    <img
      alt="Earth"
      src="https://earth.com/live.jpg"
    />
    <p>
      Earth
    </p>
  </div>
</div>
`;
